File: app/src/main/java/org/kikermo/thingsaudioreceiver/nowplaying/NowPlayingContract.java
package org.kikermo.thingsaudioreceiver.nowplaying;

import org.kikermo.thingsaudioreceiver.BasePresenter;
import org.kikermo.thingsaudioreceiver.BaseView;
import org.kikermo.thingsaudioreceiver.model.data.PlayPosition;
import org.kikermo.thingsaudioreceiver.model.data.PlayState;
import org.kikermo.thingsaudioreceiver.model.data.Track;

/**
 * Created by EnriqueR on 17/02/2017.
 */

public interface NowPlayingContract {
    public interface View extends BaseView<Presenter> {

        void updateTrack(Track track);

        void updatePosition(PlayPosition position);

        void updatePlayState(PlayState playState);

    }

    public interface Presenter extends BasePresenter {

    }
}

File: app/src/main/java/org/kikermo/thingsaudioreceiver/nowplaying/NowPlayingContract.kt
package org.kikermo.thingsaudioreceiver.nowplaying

import org.kikermo.thingsaudio.core.base.IPresenter
import org.kikermo.thingsaudio.core.base.IView
import org.kikermo.thingsaudio.core.api.model.PlayState
import org.kikermo.thingsaudio.core.api.model.Track

interface NowPlayingContract {
    interface View : IView {

        fun updateTrack(track: Track)

        fun updatePosition(position: Int)

        fun updatePlayState(playState: PlayState)
    }

    interface Presenter : IPresenter<View>
}
